### PR TITLE
Np 3066 add validation check after updating attribtues

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: cognito-pre-token-generation/src/test/java/no/unit/nva/cognito/TriggerHandlerTest.java [hunk 1]
```diff
@@ -299,6 +299,30 @@ public void handleRequestReturnsUserWithUserRoleWhenUserIsFeideHostedUserAndUser
...
+
+        handler.handleRequest(requestEvent, mock(Context.class));
+        String logMessages = logger.getMessages();
+        assertThat(logMessages, containsString(COGNITO_UPDATE_FAILURE_WARNING));
+        String logMessagesAfterWarningPrefix =
+            logMessages.substring(logMessages.indexOf(COGNITO_UPDATE_FAILURE_WARNING));
+        assertThat(logMessagesAfterWarningPrefix, containsString(CUSTOM_APPLICATION_ROLES));
+    }
+
+    private void setupTriggerHandler() {
```
Comment (axthosarouris) [inline cognito-pre-token-generation/src/test/java/no/unit/nva/cognito/TriggerHandlerTest.java:318]: moved code from `init` function for reusability

File: cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/service/UserPoolEntryUpdater.java [hunk 2]
```diff
@@ -64,6 +67,26 @@ private static AWSCognitoIdentityProvider defaultCognitoProvider() {
             .build();
     }
 
+    private void verifyThatCognitoHasUpdatedEntries(UserDetails userDetails, List<AttributeType> desiredAttributes) {
+        AdminGetUserResult response = getCognitoUserDetails(userDetails);
+        List<AttributeType> actualAttributes = response.getUserAttributes();
+        desiredAttributes.removeAll(actualAttributes);
```
Comment (janbj) [inline cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/service/UserPoolEntryUpdater.java:73]: What is desiredAttributes?

File: cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/service/UserPoolEntryUpdater.java [hunk 3]
```diff
@@ -64,6 +67,26 @@ private static AWSCognitoIdentityProvider defaultCognitoProvider() {
...
+        if (!desiredAttributes.isEmpty()) {
+            logger.warn(COGNITO_UPDATE_FAILURE_WARNING + toString(desiredAttributes));
```
Comment (janbj) [inline cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/service/UserPoolEntryUpdater.java:75]: Validation only for logging?

Comment (axthosarouris) [reply cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/service/UserPoolEntryUpdater.java]: I am only interested in logging at this point. I have not decided what to do next. I will decide after I see the logs.

File: cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/service/UserPoolEntryUpdater.java (hunk 2 above)
Comment (axthosarouris) [reply cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/service/UserPoolEntryUpdater.java]: insertedAttributes. fixed.

**Review by brinxmat — APPROVED** [review]

**Review by janbj — APPROVED** [review]

**Pull request merged**